js/script.js

```diff
@@ -332,3 +332,427 @@
+
+// console.log(removed);
+
+// Array tasks 5;
+// const array = [1, 2, 3, 4, 5];
+// array.splice(2, 0, 'a', 'b', 'c');
+
+// console.log(array);
+
+// Array tasks 6;
+// const array = [1, 2, 3, 4, 5];
+// array.splice(1, 0, 'a', 'b'), array.splice(6, 0, 'c'), array.splice(8, 0, 'e');
+
+// console.log(array);
+
+// Array tasks 7;
+// const array = [3, 4, 1, 2, 7];
+
+// console.log(array.sort());
+
+// // Array tasks 8;
+// const obj = {
+//   js: 'test',
+//   jq: 'hello',
+//   css: 'world',
+// };
+
+// console.log(Object.keys(obj));
+
+// Array tasks 9;
+// const str = '123456';
+// revStr = str.split('').reverse().join('');
+// console.log(revStr);
+
+// Array tasks 10/11;
+// const veb = 'https://docs.google.com/';
+// const str = 'index.html';
+
+// console.log(veb.indexOf('https://'));
+// console.log(str.lastIndexOf('.html'));
+
+// Array tasks 12;
+// const array = [1, 2, 3, 4, 5];
+
+// console.log(array.includes(5));
+
+// Array tasks 13;
+// let num = 31;
+// let result = false;
+// for(i = 2; i < num; i++) { 
+//   if(num % i == 0){
+//   result = true;
+//   break; 
+//   }
+// }
+// if(result == true) {
+//   console.log('true')
+// } else {
+//   console.log('false')
+// }
+
+// Array tasks 14;
+// let num = [1, 2, 3,2, 4, 5];
+// let result = false;
+//   for (let i = 1; i < num.length; i++) {
+//     if (num[i] == num[i+1]) {
+//     result = true;
+//     break;
+//     }
+//   }
+
+//   if (result == true) {
+//   console.log('yes');
+//   } else {
+//   console.log('no');
+//   }
+
+// Array tasks 15;
+// const arr = [];
+//   for (i = 1; i <= 9; i++) {
+//     str = '';
+//     for (j = 0; j < i; j++) {
+//       str += i;
+//     }
+//     arr.push(str);
+//   }
+// console.log(arr);
+
+// Function tasks 1;
+// function arrayFill(value, length) {
+//   let arr = [];
+//   for(i = 0; i < length; i++) {
+//     arr.push(value);
+//   }
+//   return arr;
+// }
+// const data = 3;
+// const valueToFill = 'a';
+// console.log(arrayFill(valueToFill, data));
+
+// Function tasks 2;
+// const data = [1, 2, 3];
+// const result = [];
+// for(i = data.length - 1; i >= 0; --i) {
+//   result.push(data[i]);
+// }
+// console.log(result);
+
+// // Function tasks 3;
+// const data = [0, 1, false, 2, undefined, '', 3, null];
+
+// function compact(arr) {
+//   return arr.filter(function(a) { 
+//     return a;
+//   });
+// }
+
+// console.log(compact(data));
+
+// Function tasks 4;
+// const data = [['a', 1], ['b', 2]];
+// const fromPairs = array => {
+// const obj = {};
+// for(i = 0; i < array.length; i++) {
+//   obj[array[i][0]] = array[i][1];
+// }
+// return obj;
+// };
+// console.log(fromPairs(data));
+
+// Function tasks 5;
+// const data = [1,2,3,2,1]
+// const without = (arr, ...args) => arr.filter((el) => !args.includes(el)); 
+// console.log(without(data, 3, 1));
+
+// Function tasks 6;
+// const data = [1, 2, 1, 2, 3];
+// function unique () {return data.filter((e,i,a) => a.indexOf(e) == i)}
```

Comment:
```suggestion
// function unique () {return data.filter((e,i,a) => a.indexOf(e) === i)}
```
!!!!
во-первых, строгое сравнение
во-вторых, что за e,i,a ??

element/item
дальше idx/index
array/arr